python/vosk/__init__.py

```diff
@@ -34,6 +46,51 @@ def __del__(self):
     def vosk_model_find_word(self, word):
         return _c.vosk_model_find_word(self._handle, word.encode('utf-8'))
 
+    def get_model_path(self, model_name, lang):
+        for directory in MODEL_DIRS:
+            if directory == None or not Path(directory).exists() and not directory == MODEL_DIRS[2]:
+                pass
+            else:
+                if MODEL_DIRS[2] == directory and not MODEL_DIRS[2].exists():
+                    MODEL_DIRS[2].mkdir()
+                model_file_list = os.listdir(directory)
+                if model_name == None:
+                    model_path = Path(directory, self.get_model_by_lang(model_file_list, directory, lang))
+                else:
+                    model_path = Path(directory, self.get_model_by_name(model_file_list, directory, model_name))
+                if not model_path.exists():
+                    urlretrieve(MODEL_PRE_PATH + str(model_path.name) + '.zip', str(model_path) + '.zip')
```

Comment:
I don't think it works as expected here. You first need to search for the model in all folders, only then download. You are trying to download for every base folder.